include/boost/gil/detail/math.hpp

```diff
@@ -28,6 +43,124 @@ inline detail::kernel_2d<T, Allocator> get_identity_kernel()
     return kernel;
 }
 
+// Please refer https://stackoverflow.com/a/10032882/14958679 for getting an overview of the
+// concept applied for obtaining higher order sobel kernels.
+
+/// \defgroup KernelGeneration
+/// \brief Contains documentation for functions used in kernel generation.
+///
+/// Separate functions are used for generating only those kernels whose dimensions are greater than
+/// 5x5. Smaller kernels are fed directly to the algorithm.
+///
+
+/// \addtogroup KernelGeneration
+/// @{
+
+/// \brief Produces higher order kernel vector by performing discrete convolution between lower
+/// order kernel vector and a smoothing kernel vector(kernel used for suppressing noise).
+/// \param kernel1 - First argument for kernel vector convolution.
+/// \param kernel2 - Second argument for kernel vector convolution.
+/// \tparam T1 - Type of first argument for kernel vector convolution.
+/// \tparam T2 - Type of second argument for kernel vector convolution.
+template<typename T1, typename T2>
+std::vector<std::vector<float>> kernel_convolve_impl(T1 kernel1, T2 kernel2)
+{
+    size_t convolved_kernel_size = kernel1.size() + kernel2.size() - 1;
+    std::vector<std::vector<float>> convolved_kernel(convolved_kernel_size,
+                                                         std::vector<float>(convolved_kernel_size)),
+                                    dummy_kernel(convolved_kernel_size,
+                                                         std::vector<float>(convolved_kernel_size));
+    
+    // 'dummy_kernel' will be made by padding 'kernel1' with appropriate no. of rows and columns 
+    // containing zeros to match the size specified by 'convolved_kernel_size'.
+    // 'convolved kernel' will store the result obtained after applying convolution between 
+    // 'dummy_kernel' and 'kernel2'.
+    // 'padding_reference' will be used to apply appropriate padding with zeros around 'kernel1'
+    // to create 'dummy_kernel'.
+    
+    std::ptrdiff_t padding_reference = (kernel2.size() - 1) / 2;
```

Comment:
I'm not sure the `reference` wording is clear here. Why not just `padding`?

Reply:
Here, ```padding``` refers to the strip of zeros created around ```kernel1``` which is required for our intended convolution. The variable ```padding_reference``` is meant to be used as a reference for finding the location of blocks from where padding starts inside ```dummy_kernel```. Shall I change the name of that variable from ```padding_reference``` to ```padding```?

Reply:
> from where padding starts 

What about `padding_origin` then?